apps/playground/src/app/modules/components/upload-module/upload-module.component.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Angular imports
+ */
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  FormControlStatus,
+  NonNullableFormBuilder,
+  Validators
+} from '@angular/forms';
+
+/**
+ * 3rd-party imports
+ */
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  templateUrl: './upload-module.component.html',
+  styleUrls: ['./upload-module.component.scss']
+})
+export class UploadModuleComponent implements OnInit, OnDestroy {
+
+  private unsubscribe$ = new Subject<void>();
+
+  inputForm = this.formBuilder.group({
+    label: ['Ajouter un fichier', Validators.required],
+    hint: 'This is a description',
+    name: 'input-name',
+    multiple: false,
+    disabled: false,
+    hasFailed: false,
+    failureMessage: 'Error message',
+  });
+
+  form = this.formBuilder.group({
+    input: ''
+  });
+
+  inputFormErrors: Record<string, string> = {};
+  selectedFiles: any;
+
+  constructor(private formBuilder: NonNullableFormBuilder) {}
+
+  ngOnInit(): void {
+    this.initForms();
+  }
+
+  private initForms(): void {
+    this.handleErrors('Please, enter a label for this Component', 'label');
```

Comment:
You need to add these 2 ones : 
```typescript
this.handleErrors(
      'Please, enter a hint message for this Component',
      'hint'
    );
    this.handleErrors(
      'Please, enter a failure message for this Component if the hasFailed attribute is true'
    );
```

Reply:
I still don't understand why we don't use Validators to display error messages